### PR TITLE
Feature/ignore invisible

### PR DESCRIPTION
**Pull request by roesslerj**

### CODE REVIEW

File: src/main/java/de/retest/web/WebData.java [hunk 1]
```diff
@@ -93,4 +93,21 @@ public int getAsInt( final String key ) {
...
 				+ (value != null ? value.getClass() : "null") + " to int!" );
 	}
+
+	public boolean isShown() {
+		final Boolean shown = (Boolean) wrappedData.get( "shown" );
+		if ( shown != null && !shown ) {
+			return false;
+		}
+		final Rectangle outline = getOutline();
+		if ( outline == null
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebData.java]: Code Smell Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbtpGavsSodgw_KFaHd&open=AWbtpGavsSodgw_KFaHd)

[ISSUE_KEY]: AWbtpGavsSodgw_KFaHd

File: src/main/java/de/retest/web/WebDataFilter.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// TODO This should be part of recheck and somehow use RecheckAdapter#convert to filter the result.
+public class WebDataFilter {
+
+	/**
+	 * Tags that will not be ignored even if they are not shown.
+	 */
+	private static final Set<String> specialTags = new HashSet<>( Arrays.asList( "head", "meta", "option", "title" ) );
+
+	public static boolean ignore( final WebData webData ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/WebDataFilter.java]: Now we can easily create unit tests for this. What do you think (especially when it comes to naming)?

File: src/main/java/de/retest/web/WebDataFilter.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package de.retest.web;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+// TODO This should be part of recheck and somehow use RecheckAdapter#convert to filter the result.
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx6&open=AWbuNaawTJjFgJLTqtx6)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx6

File: src/main/java/de/retest/web/WebDataFilter.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+public class WebDataFilter {
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx7&open=AWbuNaawTJjFgJLTqtx7)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx7

File: src/main/java/de/retest/web/WebData.java (hunk 1 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebData.java]: Code Smell Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbtpGavsSodgw_KFaHd&open=AWbtpGavsSodgw_KFaHd)

[ISSUE_KEY]: AWbtpGavsSodgw_KFaHd

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 2 above)
Comment (roesslerj) [reply src/main/java/de/retest/web/WebDataFilter.java]: Naming: shouldIgnore (Prädikat) vs. Ignore (Anweisung).

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 3 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx6&open=AWbuNaawTJjFgJLTqtx6)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx6

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 4 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx7&open=AWbuNaawTJjFgJLTqtx7)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx7

File: src/main/java/de/retest/web/WebData.java (hunk 1 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebData.java]: Code Smell Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbtpGavsSodgw_KFaHd&open=AWbtpGavsSodgw_KFaHd)

[ISSUE_KEY]: AWbtpGavsSodgw_KFaHd

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 4 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx7&open=AWbuNaawTJjFgJLTqtx7)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx7

File: src/main/java/de/retest/web/WebData.java (hunk 1 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebData.java]: Code Smell Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbtpGavsSodgw_KFaHd&open=AWbtpGavsSodgw_KFaHd)

[ISSUE_KEY]: AWbtpGavsSodgw_KFaHd

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 3 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx6&open=AWbuNaawTJjFgJLTqtx6)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx6

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 4 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx7&open=AWbuNaawTJjFgJLTqtx7)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx7

File: src/main/java/de/retest/web/WebData.java (hunk 1 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebData.java]: Code Smell Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbtpGavsSodgw_KFaHd&open=AWbtpGavsSodgw_KFaHd)

[ISSUE_KEY]: AWbtpGavsSodgw_KFaHd

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 3 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx6&open=AWbuNaawTJjFgJLTqtx6)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx6

File: src/main/java/de/retest/web/WebDataFilter.java (hunk 4 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/WebDataFilter.java]: Code Smell Code Smell: Add a private constructor to hide the implicit public one. ([squid:S1118](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=94&issues=AWbuNaawTJjFgJLTqtx7&open=AWbuNaawTJjFgJLTqtx7)

[ISSUE_KEY]: AWbuNaawTJjFgJLTqtx7

**Comment by beatngu13** [comment]
Feel free to merge if you are good with my changes.

**Pull request merged**